### PR TITLE
Study - Ruby vs Javascript

### PR DESCRIPTION
**Pull request by zchenbonterra**
-comfort: 4

I used the require reading.

### CODE REVIEW

File: study.md [hunk 1]
```diff
@@ -108,7 +117,9 @@ boomhauer.hello
 Write some JavaScript code that duplicates this functionality.
 
 ```javascript
-// your answer here
+const Person = function (name){
```
Comment (bernardlee) [inline study.md:120]:
```javascript
const Person = function(name) {
  this.name = name;
}

Person.prototype.hello = function() {
  console.log(`Hello, my name is ${this.name}.`);
}

const boomhauer = new Person('Jeff Boomhauer');

boomhauer.hello();
```

File: study.md [hunk 2]
```diff
@@ -129,7 +140,8 @@ Write some JavaScript code that uses the array `filter()` method to duplicate
 this functionality.
 
 ```javascript
-// your answer here
+let numbers = [1,2,3,4,5,6,7,8,9,10];
+let filteredNumbers = filter((value)=>x%3===0)
```
Comment (bernardlee) [inline study.md:144]: `filter` needs to be invoked on `numbers`.  The anonymous function also names `value` as its parameter but the return value uses `x`, which is undefined.

File: study.md [hunk 3]
```diff
@@ -151,5 +163,6 @@ Write some Ruby code that uses the Array `map` method and code blocks to
 duplicate this functionality.
 
 ```ruby
-# your answer here
+numbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10]
+squares = numbers.map(|x| x*x)
```
Comment (bernardlee) [inline study.md:167]: This will cause an error.  Blocks uses braces not parentheses.

**Comment by zchenbonterra** [comment]
so it should be the following?
let filteredNumbers = numbers.filter((value)=>x%3===0)

**Comment by bernardlee** [comment]
That's the correct invocation of `filter`, but you're still using `x` in the callback and that's undefined.

**Comment by zchenbonterra** [comment]
Okay, I need a one on one on this.

**Pull request closed without merging**